dds/DdsDcpsCore.idl

```diff
@@ -471,9 +474,7 @@ module DDS {
       @id(0x002E) TopicDataQosPolicy topic_data;
       @id(0x002D) GroupDataQosPolicy group_data;
       @id(0x0073) DataRepresentationQosPolicy representation;
-      /* TODO
       @id(0x0074) TypeConsistencyEnforcementQosPolicy type_consistency;
-      */
```

Comment:
Looks like this may need to get populated in SEDP?  Do we need the ParameterListConverter to be able to translate from the Parameter to the structure?

Reply:
> Looks like this may need to get populated in SEDP?

I also had to add it to `RtpsCore.idl` if that's what you mean for SEDP. The way I did it, I need the conversion for parameter list in `ParameterListConverter.cpp`.

Reply:
I'm asking how these structures get their values "populated."

ParameterListConverter has two directions of conversion:
- "to ParameterList" is for sending discovery info about endpoints in the current participant
- "from ParameterList" is for receiving discovery info about endpoints in other participants

When "this" Sedp receives an announcement on the network and transforms it "from ParameterList", where does the resulting type_consistency info end up?

One possible answer is just that we don't support this in the current iteration.  If so, it shouldn't appear in the user API.

Reply:
I haven't handled the "from" ParameterList. It was a question that came up when Mike and I discussed. There was some confusion based on the mention that the policy only applied to the DataReader. It should be easy to do the reverse to a point. The question I'll have to figure out is how far I can take it until we need more supporting code to make it useful.

Reply:
Is the "from ParameterList" more clear based on the comment above?  If not we can discuss outside of this PR review.